huaweicloud/data_source_huaweicloud_identity_custom_role_test.go

```diff
@@ -0,0 +1,77 @@
+package huaweicloud
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccHuaweiCloudIdentityCustomRoleDataSource_basic(t *testing.T) {
+	var rName = fmt.Sprintf("ACCPTTEST-%s", acctest.RandString(5))
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccPreCheckAdminOnly(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccHuaweiCloudIdentityCustomRoleDataSource_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIdentityCustomDataSourceID("data.huaweicloud_identity_custom_role.role_1"),
+					resource.TestCheckResourceAttr(
+						"data.huaweicloud_identity_custom_role.role_1", "name", rName),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIdentityCustomDataSourceID(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Can't find role data source: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("Role data source ID not set")
+		}
+
+		return nil
+	}
+}
+
+func testAccHuaweiCloudIdentityCustomRoleDataSource_basic(rName string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_identity_role" test {
+	name = "%s"
```

Comment:
please indent two spaces for each nesting level.

Reply:
sure. Sorry for not paying attention.